testing/utils/openshift_ci/pr_args.sh

```diff
@@ -89,17 +89,22 @@ else
     last_comment_page_json=$(curl -sSf "$PR_LAST_COMMENT_PAGE_URL")
 fi
 
-REQUIRED_AUTHOR_ASSOCIATION=CONTRIBUTOR
+REQUIRED_AUTHOUR=$(jq -r .user.login <<< "$pr_json")
+REQUIRED_AUTHOR_ASSOCIATION="CONTRIBUTOR"
 
 echo "PR comments URL: $PR_COMMENTS_URL" >&2
 last_user_test_comment=$(echo "$last_comment_page_json" \
-                             | jq '.[] | select(.author_association == "'$REQUIRED_AUTHOR_ASSOCIATION'") | .body' \
-                             | (grep "$test_anchor" || true) \
-                             | tail -1 | jq -r)
-
-if [[ -z "$last_user_test_comment" ]]; then
-    echo "ERROR: last comment of from a '$REQUIRED_AUTHOR_ASSOCIATION' could not be found (searching for '$test_anchor') ..." >&2
+                             | jq -r '.[-1] | .body | match("^/test.+") | .string')
+last_user_test_issuer=$(echo "$last_comment_page_json" \
+	                     | jq -r '.[-1] | .user.login')
+last_user_test_issuer_status=$(echo "$last_comment_page_json" \
+			     | jq -r '.[-1] | .user.login == "'$REQUIRED_AUTHOUR'" or .author_association == "'$REQUIRED_AUTHOR_ASSOCIATION'"')
+
+if [[ -z "$last_user_test_comment" ]] || [[ "$last_user_test_issuer" == "false" ]]; then
+    echo "ERROR: last comment of either from a '$REQUIRED_AUTHOR_ASSOCIATION' or '$REQUIRED_AUTHOUR' could not be found (searching for '$test_anchor') ..." >&2
     exit 1
+else
+    echo "INFO: last test comment '$last_user_test_comment' issued by $last_user_test_issuer"
 fi
```

Comment:
```suggestion
    exit 1
fi
echo "INFO: last test comment '$last_user_test_comment' issued by $last_user_test_issuer"
```
no need for `else` after `exit` or `return`

Reply:
hmmm, sure no commands will be executed after  `exit` command and the control will be given to the caller function after a `return` command. 

If you mean to say `else` block will not be executed then you are wrong because bash does execute the statements.
```
[cmusali@cmusali topsail]$ a=10

[cmusali@cmusali topsail]$ if [[ $a -eq 11 ]];then echo 'good';exit 0;else echo 'bad';fi
bad
```

The `else` block does nothing here other than providing some insightful information to the user. I can remove that if you believe that is not required here.

Reply:
what I meant is that this pattern:
```
if (test); then
  statement1;
  exit # or return
else
  statement2;  
fi
```
should be avoided, as this one is simpler and strictly equivalent:
```
if (test); then
  statement1;
  exit # or return
fi
statement2;  
```

it's linked to the guard pattern: https://en.wikipedia.org/wiki/Guard_(computer_science)

Reply:
hmmm. make sense. thanks for the feedback